packages/lib/CalendarService.ts

```diff
@@ -251,9 +251,13 @@ export default abstract class BaseCalendarService implements Calendar {
     objects.forEach((object) => {
       if (object.data == null) return;
 
-      const jcalData = ICAL.parse(object.data);
+      const jcalData = ICAL.parse(sanitizeCalendarObject(object));
```

Comment:
Depending on the CalDAV service provider, the `object.data` may or may not have `\r` or `\r\n` as the EOL. `ICAL.parse` expects it to have '\r\n' to work as expected. Also, the `object.data` received from the `fetchCalendarObjects` function is at times faulty in placing the EOL, which is why we need to make sure that it doesn't break the structure of expected input for `ICAL.parse`, hence the need for sanitization.